cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -993,7 +1027,7 @@ CacheAllocator<CacheTrait>::releaseBackToAllocator(Item& it,
           XDCHECK(ReleaseRes::kReleased != res);
           res = ReleaseRes::kRecycled;
         } else {
-          allocator_[tid]->free(head);
```

Comment:
Do we still need the logic at line 982 (I couldn't add a comment there)? It seems to me that we no longer mark the child items are moving, so this will never be true?